service/lib/dinstaller/dbus/software_service.rb

```diff
@@ -71,10 +71,19 @@ def service
       # @return [Array<::DBus::Object>]
       def dbus_objects
         @dbus_objects ||= [
-          DInstaller::DBus::Software::Manager.new(@backend, logger),
+          dbus_software_manager,
           DInstaller::DBus::Software::Proposal.new(logger)
         ]
       end
+
+      # @return [DInstaller::DBus::Software::Manager]
+      def dbus_software_manager
+        return @dbus_software_manager if @dbus_software_manager
+
+        @backend.on_progress_change { dispatch }
```

Comment:
NP: Maybe it would be better to configure this in the constructor, while instantiating the backend.

Reply:
Another option is to define a `#start` method containing that kind of configurations, similar to manager service.

Reply:
OK, it makes sense. After all, we are calling the `register_*` methods in the constructor.